src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -1336,6 +1336,50 @@ def test_fw_binaryoperator_matmul(fortran_writer, tmpdir, fortran_reader):
     assert Compile(tmpdir).string_compiles(result)
 
 
+def test_fw_binaryoperator_bounds(tmpdir, fortran_reader, fortran_writer):
+    '''Check the FortranWriter class binary_operation method only writes
+    out the array name when the array reference is within a lbound or
+    ubound binary operation.
+
+    '''
+    code = (
+        "program test\n"
+        "  real, dimension(10,10,10) :: a\n"
+        "  real, dimension(10,10,10) :: b\n"
+        "  real, dimension(10,10,10) :: c\n"
+        "  integer :: idx\n"
+        "  integer :: idx_1\n\n"
+        "  do idx = LBOUND(a(1,1,1), 3), UBOUND(a(:,:,:), 3), 1\n"
```

Comment:
What about e.g. `UBOUND(a(1:3), 1)` where `a` actually has an extent `> 3`? The `datatype` property that I'm adding in #1850 will allow this to be sorted out.

Reply:
This test has been removed. The transformation does not modify arrays within LBOUND and UBOUND operators.